x/cdp/keeper/cdp.go

```diff
@@ -410,6 +413,28 @@ func (k Keeper) CalculateCollateralToDebtRatio(ctx sdk.Context, collateral sdk.C
 	return collateralBaseUnits.Quo(debtTotal)
 }
 
+// LoadAugmentedCDP creates a new augmented CDP from an existing CDP
+func (k Keeper) LoadAugmentedCDP(ctx sdk.Context, cdp types.CDP) (types.AugmentedCDP, sdk.Error) {
+	// calculate collateralization ratio
+	collateralizationRatio, err := k.CalculateCollateralizationRatio(ctx, cdp.Collateral, cdp.Principal, cdp.AccumulatedFees)
+	if err != nil {
+		return types.AugmentedCDP{}, err
+	}
+	// calculate collateral value in debt coin
+	var totalDebt int64
+	if len(cdp.AccumulatedFees) > 0 {
+		totalDebt += cdp.AccumulatedFees[0].Amount.Int64()
+	}
+	totalDebt += cdp.Principal[0].Amount.Int64()
```

Comment:
CDPs can contain more than one principal/fee denom, so this could miss out some. They're all treated as equal to one debt coin.

Reply:
https://github.com/Kava-Labs/kava/pull/347/commits/fdee0711103958ee0661f67808090297e1f95f6f

Reply:
https://github.com/Kava-Labs/kava/pull/347/commits/fdee0711103958ee0661f67808090297e1f95f6f